lib/install/install.go

```diff
@@ -110,7 +111,7 @@ func (i *Installer) Execute(req *installpb.ExecuteRequest, stream installpb.Agen
 		case result := <-i.execDoneC:
 			if result.err != nil {
 				// Phase finished with an error
-				return status.Error(codes.FailedPrecondition, result.err.Error())
+				return status.Error(codes.Aborted, result.err.Error())
```

Comment:
To be frank, I still have no idea why would one use one error code versus another, and it seems to be vital to know where to use what because it affects the logic of the install service AFAIU.

Can these generic codes maybe be adopted to our needs? E.g. have our own defined errors wrapping these ones, with clear names/descriptions and use them everywhere we need to return an error which affects the service behavior. We already have `Abort`, maybe also have `StopWithoutClean` (totally made up) or something.... you get the idea.

Reply:
Yes, but on gRPC boundary all errors are wrapped in gRPC error.

Reply:
Yeah, that makes sense. At the very least we should probably document what impact each returned code has on the service (if it's not yet).